datafusion/optimizer/src/decorrelate_where_exists.rs

```diff
@@ -670,4 +678,56 @@ mod tests {
 
         assert_plan_eq(&plan, expected)
     }
+
+    #[test]
+    fn exists_subquery_non_distinct_aggregate() -> Result<()> {
+        let table_scan = test_table_scan()?;
+        let subquery_scan = test_table_scan_with_name("sq")?;
+
+        let subquery = LogicalPlanBuilder::from(subquery_scan)
+            .filter(col("sq.a").gt(col("test.b")))?
+            .project(vec![col("sq.a"), col("sq.c")])?
+            .aggregate(vec![col("a"), col("c")], vec![count(col("a"))])?
+            .build()?;
+        let plan = LogicalPlanBuilder::from(table_scan)
+            .filter(exists(Arc::new(subquery)))?
+            .project(vec![col("test.b")])?
+            .build()?;
+
+        // Should not be optimized to join.
+        let expected = "Projection: test.b [b:UInt32]\
+                        \n  Filter: EXISTS (<subquery>) [a:UInt32, b:UInt32, c:UInt32]\
+                        \n    Subquery: [a:UInt32, c:UInt32, COUNT(sq.a):Int64;N]\
+                        \n      Aggregate: groupBy=[[sq.a, sq.c]], aggr=[[COUNT(sq.a)]] [a:UInt32, c:UInt32, COUNT(sq.a):Int64;N]\
+                        \n        Projection: sq.a, sq.c [a:UInt32, c:UInt32]\
+                        \n          Filter: sq.a > test.b [a:UInt32, b:UInt32, c:UInt32]\
+                        \n            TableScan: sq [a:UInt32, b:UInt32, c:UInt32]\
+                        \n    TableScan: test [a:UInt32, b:UInt32, c:UInt32]";
+
+        assert_plan_eq(&plan, expected)
+    }
+
+    #[test]
+    fn exists_subquery_aggragte_distinct() -> Result<()> {
+        let table_scan = test_table_scan()?;
+        let subquery_scan = test_table_scan_with_name("sq")?;
+
+        let subquery = LogicalPlanBuilder::from(subquery_scan)
+            .filter(col("sq.a").gt(col("test.b")))?
+            .project(vec![col("sq.a"), col("sq.c")])?
```

Comment:
Can you add a test for:
1. When there is no projection in the subquery?
2. When the projection in the subquery is an expression (like `sq.a + sq.b`)?